Client/Frontend/Browser/OpenPdfHelper.swift

```diff
@@ -91,11 +132,37 @@ class OpenPdfInHelper: NSObject, OpenInHelper, UIDocumentInteractionControllerDe
     private var docController: UIDocumentInteractionController? = nil
     private var openInURL: NSURL?
 
-    lazy var openInView: OpenInView = getOpenInView(self)()
+    lazy var openInView: OpenInView? = getOpenInView(self)()
+
+    var mimeType: MimeType = .PDF
+    var fileType: FileType = .PDF
 
-    init(url: NSURL) {
+    lazy var documentDirectory: NSURL = {
+        return NSURL(string: NSTemporaryDirectory())!.URLByAppendingPathComponent(self.fileType.rawValue)
+    }()
+
+    private var filepath: NSURL?
+
+    required init(url: NSURL) {
         self.url = url
         super.init()
+        setFilePath(url.lastPathComponent ?? "file.pdf")
+    }
+
+    required init?(response: NSHTTPURLResponse) {
+        guard let responseURL = response.URL else { return nil }
+        url = responseURL
+        super.init()
+        setFilePath(response.suggestedFilename ?? url.lastPathComponent ?? "file.pdf")
```

Comment:
Ok. Gotcha.

I'm bothered by a lot of the code duplication around this constructors. I'm not convinced that both of them are needed.